### PR TITLE
Minor improvements to Player.ts

### PR DESCRIPTION
**Pull request by nwai90**

### CODE REVIEW

File: src/Player.ts [hunk 1]
```diff
@@ -379,12 +376,7 @@ export class Player implements ISerializable<SerializedPlayer> {
 
     // Manutech hook
     if (amount > 0 && this.corporationCard !== undefined && this.corporationCard.name === CardName.MANUTECH) {
-      if (resource === Resources.MEGACREDITS) this.megaCredits += amount;
-      if (resource === Resources.STEEL) this.steel += amount;
-      if (resource === Resources.TITANIUM) this.titanium += amount;
-      if (resource === Resources.PLANTS) this.plants += amount;
-      if (resource === Resources.ENERGY) this.energy += amount;
-      if (resource === Resources.HEAT) this.heat += amount;
+      Manutech.resolveCorpEffect(this, resource, amount);
```
Comment (Lynesth) [inline src/Player.ts:379]: onProductionGain ?

**Review by Lynesth — CHANGES_REQUESTED** [review]

**Review by Lynesth — APPROVED** [review]

File: src/Player.ts [hunk 2]
```diff
@@ -1605,22 +1614,23 @@ export class Player implements ISerializable<SerializedPlayer> {
...
   public canAfford(cost: number, canUseSteel: boolean = false, canUseTitanium: boolean = false, canUseFloaters: boolean = false, canUseMicrobes : boolean = false): boolean {
-    return cost <= this.spendableMegacredits() +
-      (canUseSteel ? this.steel * this.steelValue : 0) +
-      (canUseTitanium ? this.titanium * this.getTitaniumValue() : 0) +
-      (canUseFloaters ? this.getFloatersCanSpend() * 3 : 0) +
-      (canUseMicrobes ? this.getMicrobesCanSpend() * 2 : 0);
+    return cost <= this.getTotalSpendableResources(canUseSteel, canUseTitanium, canUseFloaters, canUseMicrobes);
+  }
+
+  private getTotalSpendableResources(canUseSteel: boolean, canUseTitanium: boolean, canUseFloaters: boolean, canUseMicrobes: boolean, spendableSteel?: number, spendableTitanium?: number): number {
```
Comment (sienmich) [inline src/Player.ts:1626]: This conflicts with my PR: https://github.com/bafolts/terraforming-mars/pull/2559
Could you maybe leave this part for now? I'm planning to further improve it

**Review by sienmich — COMMENTED** [review]
I like all those changes, putting constants in constants is good. I was just working on similar thing and we'll have a conflict.

**Review by kberg — COMMENTED** [review]
> I like all those changes, putting constants in constants is good. I was just working on similar thing and we'll have a conflict.

What's the state of this PR? Just seeing if there is something I can do.

**Review by sienmich — APPROVED** [review]

File: src/Player.ts [hunk 3]
```diff
@@ -1099,11 +1091,31 @@ export class Player implements ISerializable<SerializedPlayer> {
...
+  private canUseFloaters(card: ICard): boolean {
+    return card.tags.includes(Tags.VENUS);
+  }
+
+  private getMcTradeCost(): number {
+    return 9 - this.colonyTradeDiscount;
+  }
+
+  private getEnergyTradeCost(): number {
+    return 3 - this.colonyTradeDiscount;
```
[the remark this answers is not in the document]
Comment (nwai90) [reply src/Player.ts]: Done

File: src/Player.ts (hunk 2 above)
Comment (kberg) [reply src/Player.ts]: @sienmich this is OK?

Comment (kberg) [reply src/Player.ts]: Sorry never mind, I'm all caught up.

**Review by kberg — APPROVED** [review]

**Pull request merged**